src/units/abilities.cpp

```diff
@@ -867,41 +867,123 @@ static void add_name(std::string& weapon_abilities, bool active, const config::a
 std::string attack_type::weapon_specials(bool only_active, bool is_backstab) const
 {
 	//log_scope("weapon_specials");
+	return weapon_specials_impl(only_active, is_backstab);
+}
+
+std::string attack_type::weapon_specials_value(const std::set<std::string> checking_tags, bool only_active, bool is_backstab) const
+{
+	//log_scope("weapon_specials_value");
+	return weapon_specials_impl(only_active, is_backstab, checking_tags, true);
+}
+
+std::string attack_type::weapon_specials_impl(bool only_active, bool is_backstab, const std::set<std::string> checking_tags, bool is_tooltip) const
+{
 	std::string res;
+	bool tag_checked;
```

Comment:
Why is this declared here? I didn't read the code carefully, but it looks like it's only used in various loops (eg, the one immediately after this) where it's unconditionally assigned before using… so I think it would make more sense to declare it separately in each loop where it's used.

Reply:
it is fixed